statefun-sdk-go/v3/pkg/statefun/context.go

```diff
@@ -64,7 +66,7 @@ type Context interface {
 }
 
 type statefunContext struct {
-	sync.Mutex
+	*sync.Mutex
```

Comment:
This was changed to a pointer so that different statefunContext instances can reference the same sync.Mutex, as opposed to each having their own sync.Mutex. I thought this was necessary because derived statefunContext instances all reference the same response object, and the mutex is used to prevent concurrent modifications to the response object?

Reply:
You are correct. Thank you.